sprocs/instance_questions_update_score.sql

```diff
@@ -33,8 +34,8 @@ BEGIN
     -- ##################################################################
     -- get the assessment_instance, max_points, and (possibly) submission_id
 
-    SELECT        s.id,                iq.id,                  ai.id, aq.max_points
-    INTO submission_id, instance_question_id, assessment_instance_id,    max_points
+    SELECT        s.id,                iq.id,                  ai.id, aq.max_points, u.uid
+    INTO submission_id, instance_question_id, assessment_instance_id,    max_points, found_uid
```

Comment:
Is this safe for group assessments? I assume so, but I haven't traced it carefully to check.

Reply:
Good point. I just tested, and the update happens as usual in this case. The fact that `found_uid` will be NULL for these assessments means that `found_uid != arg_uid` (in the IF that checks for this value in line 67) will be NULL (equivalent to false), so regardless of if there is a `uid` column in the file the exception will never be raised. The download file for manual grading doesn't include a UID column for group assessments, so this would only take place if the instructor explicitly includes a `uid` column, which would lead me to believe that the instructor is expecting the safeguard. So options are:

1. if there is a UID column in a group assessment, trigger an error always (since no assessment is linked to a single student);
2. for group assessments check the `group_name` column instead;
3. for group assessments, check if at least one user in the group matches the uid.

Option 1 is the simplest. Option 2 is probably the best, but would require either refactoring the sproc and its calls to add a group name, or repurposing the `arg_uid` to mean `arg_uid_or_group_name` instead.

Reply:
Option 2 implemented.

Reply:
Excellent, that's definitely the best-but-painful option, so thanks for doing it.